### DISCUSSION TITLE
How should clients pay for licenses & how do approvals work?

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/console#69] [Craft Console]
I'm trying to figure out how I can help a client purchase some licenses and establish a process going forward - at first I thought I could add the licenses to the organization's cart and then let the client complete the purchase; however, the cart doesn't appear to be shared across user accounts, as the client was unable to see them. I confirmed this myself afterwards.

I decided to test the new "approval" feature next. I added one of my colleagues to our organization as a user and asked him to go and try to renew an expired plugin license. He was unable to see the "Renew your license" button on the license page, so I promoted him to an admin. As an admin, he was able to add the license to his cart and go through the checkout; however, I was not prompted for approval of the purchase, it simply charged my saved card. The instructions for approval are a bit unclear - it implies that "owners" will be prompted for approval, but it also says "members" will be required to get approval:
> Members must request approval by the organization owner (you) for any purchases.

It seems like the reality is somewhere in-between?

So, here's where I am now - as part of our agreement, my organization purchased plugins we needed for development; however, the client is responsible for the core CMS license. I paid for the plugin licenses but still need the client to pay for the CMS license. Should I instruct the client to purchase through the CMS license warning banner, should I have them add a payment method and then purchase on their behalf, or is there a better way I'm not accounting for?

*Ideally* - I'd like to be able to send a link the client that goes straight to a checkout page with the licenses in the cart so they don't need to fuss with it.

**Comment by lindseydiloreto** [comment]
> _Ideally_ - I'd like to be able to send a link the client that goes straight to a checkout page with the licenses in the cart so they don't need to fuss with it.

This would be absolutely brilliant. It would save both us and the client a lot of headache.

**Comment by brandonkelly** [comment]
> at first I thought I could add the licenses to the organization's cart and then let the client complete the purchase; however, the cart doesn't appear to be shared across user accounts, as the client was unable to see them.

Correct, org carts are still “yours”; they’re just optimized for purchasing things in relation to the org.

> I added one of my colleagues to our organization as a [member] and asked him to go and try to renew an expired plugin license. He was unable to see the "Renew your license" button on the license page…

That feels like something we overlooked. Will discuss internally.

> …so I promoted him to an admin. As an admin, he was able to add the license to his cart and go through the checkout; however, I was not prompted for approval of the purchase, it simply charged my saved card. The instructions for approval are a bit unclear - it implies that "owners" will be prompted for approval, but it also says "members" will be required to get approval:
> 
> > Members must request approval by the organization owner (you) for any purchases.
> 
> It seems like the reality is somewhere in-between?

Only _non-admin_ members must request approval. We’ll look into clarifying that.

> So, here's where I am now - as part of our agreement, my organization purchased plugins we needed for development; however, the client is responsible for the core CMS license. I paid for the plugin licenses but still need the client to pay for the CMS license. Should I instruct the client to purchase through the CMS license warning banner, should I have them add a payment method and then purchase on their behalf, or is there a better way I'm not accounting for?

As a starting point, ensure that the Craft and plugin licenses are all owned by the _client’s_ organization. (When you transfer a Craft license, any associated plugin licenses that are already owned by the originating account/org will follow along with the Craft license transfer by default.)

Then ensure your client is the owner of the org, and that you’re a member. Have your client enter their billing info at `/settings/billing`, and select that payment method at `/orgs/<handle>/settings/billing`.

From there you will be able to upgrade the Craft CMS license to Pro and have your client’s card be charged for it. The purchase may or may not require your client’s approval depending on how they configured the org’s billing settings (and your member role in their org).

> _Ideally_ - I'd like to be able to send a link the client that goes straight to a checkout page with the licenses in the cart so they don't need to fuss with it.

We’ve considered this, but it feels awkward for clients to be expected to pay for a license that they don’t actually own in the first place.
